src/compute/src/sink/persist_sink.rs

```diff
@@ -452,16 +453,10 @@ where
                         }
                     }
                 }
-                Some(event) = persist_feedback_input.next() => {
-                    match event {
-                        Event::Data(_cap, _data) => {
-                            // This input produces no data.
-                            continue;
-                        }
-                        Event::Progress(frontier) => {
-                            persist_frontier = frontier;
-                        }
-                    }
+                (write_handle, upper) = &mut upper_future => {
+                    persist_frontier = upper;
+                    let fut = gen_upper_future(persist_frontier.clone(), write_handle);
+                    upper_future = fut;
```

Comment:
Would it be possible to replace the `upper_future` with a stream instead that calls the future's logic in a loop?

Reply:
will do! 👌